clients/scheduler/interval_action_test.go

```diff
@@ -0,0 +1,438 @@
+	Modified:   321,
+	Origin:     321,
+	Name:       "testNombre",
+	Interval:   "321",
+	Parameters: "321",
+	Target:     "testName",
+	Protocol:   "321",
+	HTTPMethod: "post",
+	Address:    "127.0.0.1",
+	Port:       3000,
+	Path:       "321",
+	Publisher:  "321",
+	User:       "321",
+	Password:   "321",
+	Topic:      "321",
+}
+
+func TestIntervalActionRestClient_Add(t *testing.T) {
+	ts := testHttpServer(t, http.MethodPost, clients.ApiIntervalActionRoute)
+
+	defer ts.Close()
+
+	url := ts.URL + clients.ApiIntervalActionRoute
+
+	params := types.EndpointParams{
+		ServiceKey:  clients.SupportSchedulerServiceKey,
+		Path:        clients.ApiIntervalActionRoute,
+		UseRegistry: false,
+		Url:         url,
+		Interval:    clients.ClientMonitorDefault,
+	}
+
+	iac := NewIntervalActionClient(params, MockEndpoint{})
+
+	res, err := iac.Add(&testIntervalAction1, context.Background())
+
+	if res == "" {
+		t.Fatal("unexpected empty string response")
+	}
+
+	if err != nil {
+		t.Fatalf("unexpected error %s", err.Error())
+	}
+}
+
+func TestIntervalActionRestClient_Delete(t *testing.T) {
+	ts := testHttpServer(t, http.MethodDelete, clients.ApiIntervalActionRoute)
+
+	defer ts.Close()
+
+	url := ts.URL + clients.ApiIntervalActionRoute
+
+	params := types.EndpointParams{
+		ServiceKey:  clients.SupportSchedulerServiceKey,
+		Path:        clients.ApiIntervalActionRoute,
+		UseRegistry: false,
+		Url:         url,
+		Interval:    clients.ClientMonitorDefault,
+	}
+
+	ic := NewIntervalActionClient(params, MockEndpoint{})
+
+	err := ic.Delete(testID1, context.Background())
+
+	if err != nil {
+		t.Fatalf("unexpected error %s", err.Error())
+	}
+}
+
+func TestIntervalActionRestClient_DeleteByName(t *testing.T) {
+	ts := testHttpServer(t, http.MethodDelete, clients.ApiIntervalActionRoute)
+
+	defer ts.Close()
+
+	url := ts.URL + clients.ApiIntervalActionRoute
+
+	params := types.EndpointParams{
+		ServiceKey:  clients.SupportSchedulerServiceKey,
+		Path:        clients.ApiIntervalActionRoute,
+		UseRegistry: false,
+		Url:         url,
+		Interval:    clients.ClientMonitorDefault,
+	}
+
+	ic := NewIntervalActionClient(params, MockEndpoint{})
+
+	err := ic.DeleteByName(testIntervalAction1.Name, context.Background())
+
+	if err != nil {
+		t.Fatalf("unexpected error %s", err.Error())
+	}
+}
+
+func TestIntervalActionRestClient_IntervalAction(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+
+		if r.Method != http.MethodGet {
+			t.Fatalf("expected http method is GET, active http method is : %s", r.Method)
+		}
+
+		expectedURL := clients.ApiIntervalActionRoute + "/" + testID1
+		if r.URL.EscapedPath() != expectedURL {
+			t.Fatalf("expected uri path is %s, actual uri path is %s", expectedURL, r.URL.EscapedPath())
+		}
+
+		data, err := json.Marshal(testIntervalAction1)
+		if err != nil {
+			t.Fatalf("marshaling error: %s", err.Error())
+		}
+		w.Write(data)
```

Comment:
Sorry, I missed this before. Can we explicitly ignore the returned values?

```go
_, _ = w.Write(data)
```

This applies for the other tests which use this method.